source/common/stats/tag_extractor_impl.h

```diff
@@ -23,13 +24,11 @@ class TagExtractorImpl : public TagExtractor {
    * @return TagExtractorPtr newly constructed TagExtractor.
    */
   static TagExtractorPtr createTagExtractor(const std::string& name, const std::string& regex,
-                                            const std::string& substr = "");
+                                            const std::string& substr = "", bool is_re2 = false);
```

Comment:
I think it should be possible to treat this as an implementation detail by looking for directives that RE2 doesn't support.

Reply:
That would be cool but I suspect there exist pattern strings that are legal in both re2 and std::regex but will be interpreted differently. Until I find proof to the contrary it's safer to capture programmer intent.